### ISSUE TITLE
Drag to dismiss the calendar contextual menu

### ISSUE THREAD

**Issue by daividssilverio** [toggl/mobileapp#6140]
The idea is to use the `BottomSheetBehaviour` to let user close the contextual menu by dragging it down to dismiss.

> Forcing the user to dismiss only by pressing the X is not good. We should allow swiping down (or dismiss the contextual menu when pressing anywhere that is not a calendar item)

_Originally posted on https://github.com/toggl/mobileapp/pull/6099#pullrequestreview-300811588 by @heytherewill

**Comment by heytherewill** [comment]
Daivid was able to quickly fix implement this in kotlin using material components, but there's a pretty weird bug that happens in our app. The WIP code is [here](https://github.com/toggl/mobileapp/tree/wip-wip-calendar-bottom-sheet). I'm putting this on hold as depending on #5829 for the time being
